etl/afd_ems_import/incident_import.py

```diff
@@ -0,0 +1,220 @@
+    file_object.seek(0)
+    return file_object.read().decode("utf-8")
+
+
+def get_data_from_email_str(source, email_str):
+    """Parse raw email content and return the attachment as a list of dictionaries
+
+    Args:
+        source (str): The data source: `afd` or `ems`
+        email_str (str): the decoded email string
+
+    Returns:
+        list: list of row dictionaries of the parsed CSV attachment
+    """
+    logging.info("Getting attachment from email...")
+    message = email.message_from_string(email_str)
+    attachment = message.get_payload()[1]
+    attachment_data = BytesIO(attachment.get_payload(decode=True))
+
+    logging.info("Parsing attachment data...")
+    if source == "ems":
+        reader = csv.DictReader(TextIOWrapper(attachment_data, encoding="iso-8859-1"))
+        return [row for row in reader]
+    elif source == "afd":
+        from pandas import read_excel
+
+        data = read_excel(
+            attachment_data, header=0, dtype={"Inc_Time": str, "Inc_Date": str}
+        )
+        return data.to_dict("records")
+
+
+def archive_email(email_obj_key):
+    """Move email fom ./inbox to ./archive
+
+    Args:
+        email_obj_key (str): the s3 object file key of the email file to be archved
+    """
+    new_key = email_obj_key.replace("inbox", "archive")
+    logging.info(f"Archiving {email_obj_key}")
+    s3_resource.meta.client.copy(
+        {"Bucket": BUCKET_NAME, "Key": email_obj_key}, BUCKET_NAME, new_key
+    )
+    s3_client.delete_object(Bucket=BUCKET_NAME, Key=email_obj_key)
+
+
+def lower_case_keys(data):
+    """Copy a list of row dicts by making each key lower case"""
+    return [{key.lower(): value for key, value in row.items() if key} for row in data]
+
+
+def set_empty_strings_to_none(data):
+    """Traverse every row property and set '', '-', and 'None' to None"""
+    for row in data:
+        for key, val in row.items():
+            if val == "" or val == "None" or val == "-":
+                row[key] = None
+
+
+def rename_columns(data, cols_to_rename):
+    """Rename columns in a list of row dictionaries.
+
+    Args:
+        data (list): the data to be processed
+        cols_to_rename (dict): a dict in the format { <old-name>: <new-name> }
+    """
+    for row in data:
+        for source_key, target_key in cols_to_rename.items():
+            row[target_key] = row.pop(source_key)
+
+
+def listify_apd_incident_numbers(data):
+    """Parse the apd_incident_numbers str into a list. This is only needed for EMS
+    records"""
+    for row in data:
+        incident_nums = row["apd_incident_numbers"]
+        row["unparsed_apd_incident_numbers"] = incident_nums
+        if incident_nums:
+            row["apd_incident_numbers"] = (
+                incident_nums.replace("-", "").replace(" ", "").split(",")
+            )
+
+
+def listify_ems_incident_numbers(data):
+    """Parse the EMS incident numbers into a list. This is only needed for AFD records.
```

Comment:
This is only needed for AFD records? Is it to make the AFD records match the EMS incident numbers? Or is the AFD part a typo?

Reply:
Surprisingly, not a typo. The AFD records have a column for EMS incident numbers, so that they can theoretically by joined to EMS records. The EMS records do not have a column for AFD incident numbers.